vendor/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -452,10 +453,14 @@ func (s *sharedIndexInformer) HandleDeltas(obj interface{}) error {
 	// from oldest to newest
 	for _, d := range obj.(Deltas) {
 		switch d.Type {
-		case Sync, Added, Updated:
+		case Sync, Added, Updated, Replaced:
 			isSync := d.Type == Sync
 			s.cacheMutationDetector.AddObject(d.Object)
 			if old, exists, err := s.indexer.Get(d.Object); err == nil && exists {
+				// If the delta is from a relist, and the RV hasn't changed, elide it.
+				if d.Type == Replaced && isObjectSame(old, d.Object) {
```

Comment:
this is an optimization. Better move it into its own commits, up for discussion.